src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java

```diff
@@ -135,14 +136,9 @@ public static FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
         return field;
     }
 
-    public static void addStoredFieldForVectorField(
-        ParseContext context,
-        FieldType fieldType,
-        String mapperName,
-        String vectorFieldAsString
-    ) {
+    public static void addStoredFieldForVectorField(ParseContext context, FieldType fieldType, String mapperName, Field vectorField) {
         if (fieldType.stored()) {
-            context.doc().add(new StoredField(mapperName, vectorFieldAsString));
+            context.doc().add(new StoredField(mapperName, vectorField.toString()));
```

Comment:
so the idea is - we only covert to String if field is stored, only always. That's good optimization!

Reply:
Yes that is correct and if you look into the issue you can see the time it takes to convert a field to string.